baseplate/integration/pyramid.py

```diff
@@ -69,9 +77,12 @@ class BaseplateConfigurator(object):
     """
 
     # TODO: remove the default on trust_trace_headers once apps are updated
-    def __init__(self, baseplate, trust_trace_headers=False):
+    def __init__(self, baseplate, trust_trace_headers=False,
+                 auth_factory=None):
         self.baseplate = baseplate
         self.trust_trace_headers = trust_trace_headers
+        self.auth_factory = auth_factory if auth_factory \
+            else AuthenticationContextFactory()
```

Comment:
Yeah that is probably better, I was not sure if I wanted to do more for the conditional check on the param but that is probably YAGNI...